src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -108,14 +111,21 @@ public TexlBinding BindBody(INameResolver nameResolver, IBinderGlue documentBind
                 throw new ArgumentNullException(nameof(documentBinderGlue));
             }
 
+            _symbols = nameResolver as ReadOnlySymbolTable;
+
             bindingConfig = bindingConfig ?? new BindingConfig(this._isImperative);
-            _binding = TexlBinding.Run(documentBinderGlue, UdfBody, UserDefinitionsNameResolver.Create(nameResolver, _args), bindingConfig, features: features, rule: rule);
+            _binding = TexlBinding.Run(documentBinderGlue, UdfBody, UserDefinitionsNameResolver.Create(nameResolver, _args, _fvArgs), bindingConfig, features: features, rule: rule);
 
             CheckTypesOnDeclaration(_binding.CheckTypesContext, _binding.ResultType, _binding);
 
             return _binding;
         }
 
+        public void SetFormulaValues(FormulaValue[] formulaValues)
```

Comment:
This could be an issue since we will need to reuse a binding across multiple invokes.

Reply:
Removed.